libp2p/protocols/pubsub/timedcache.nim

```diff
@@ -78,7 +79,18 @@ func put*[K](t: var TimedCache[K], k: K, now = Moment.now()): bool =
   # Puts k in cache, returning true if the item was already present and false
   # otherwise. If the item was already present, its expiry timer will be
   # refreshed.
+  func ensureSizeBound(t: var TimedCache[K]) =
+    if t.maxSize > 0 and t.entries.len() >= t.maxSize and k notin t:
```

Comment:
we could also always insert and then remove the oldest if `len == maxSize + 1`, in this way we don't need the expensive `k notin t`.

Reply:
actually `k notin t` uses a HashSet, so it's cheap